src/mono/wasm/runtime/web-socket.ts

```diff
@@ -336,16 +332,10 @@ function _mono_wasm_web_socket_send_buffering(ws: WebSocketExtension, buffer_vie
         }
         if (message_type === 0) {
             // text, convert from UTF-8 bytes to string, because of bad browser API
-            if (_text_decoder_utf8 === undefined) {
-                // we do not validate outgoing data https://github.com/dotnet/runtime/issues/59214
-                _text_decoder_utf8 = new TextDecoder("utf-8", { fatal: false });
-            }
 
-            // See https://github.com/whatwg/encoding/issues/172
-            const bytes = typeof SharedArrayBuffer !== "undefined" && buffer instanceof SharedArrayBuffer
-                ? (<any>buffer).slice(0, offset)
-                : buffer.subarray(0, offset);
-            return _text_decoder_utf8.decode(bytes);
+            const bytes = copyBufferIfNecessary(buffer, 0 as any, offset as any);
```

Comment:
there is type confusion here and in the original code - subarray is a method of ArrayView, which implies buffer is a Uint8Array or other type, not an ArrayBuffer. Which means `buffer instanceof SharedArrayBuffer` would never be true.

I think the operation we actually want here is named something like `ensureViewHasNonSharedBuffer`.

Reply:
the `buffer` in WS code is always instance of `Uint8Array` and the code is correct, I will rename `copyBufferIfNecessary` to `viewOrCopy`

Reply:
name `buffer` in WS is bit misleading, but most of the time it's dedicated buffer, not view of linear memory.